tests/integration/gha-run.sh

```diff
@@ -0,0 +1,153 @@
+    cat "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
+    cat "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml" | grep "${DOCKER_REGISTRY}/${DOCKER_REPO}:${DOCKER_TAG}" || die "Failed to setup the tests image"
+
+    kubectl apply -f "${tools_dir}/packaging/kata-deploy/kata-rbac/base/kata-rbac.yaml"
+    if [ "${platform}" = "tdx" ]; then
+        kubectl apply -k "${tools_dir}/packaging/kata-deploy/kata-deploy/overlays/k3s"
+    else
+        kubectl apply -f "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
+    fi
+    kubectl -n kube-system wait --timeout=10m --for=condition=Ready -l name=kata-deploy pod
+    kubectl apply -f "${tools_dir}/packaging/kata-deploy/runtimeclasses/kata-runtimeClasses.yaml"
+
+    # This is needed as the kata-deploy pod will be set to "Ready" when it starts running,
+    # which may cause issues like not having the node properly labeled or the artefacts
+    # properly deployed when the tests actually start running.
+    if [ "${platform}" = "aks" ]; then
+        sleep 240s
+    else
+        sleep 60s
+    fi
+
+    pushd "${integration_dir}/kubernetes"
+    bash setup.sh
+    bash run_kubernetes_tests.sh
+    popd
+}
+
+function cleanup() {
+    platform="${1}"
+
+    if [ "${platform}" = "aks" ]; then
+        az aks delete \
+            -g "kataCI" \
+            -n "${GH_PR_NUMBER}-${GH_PR_SHA}-${KATA_HYPERVISOR}-amd64" \
+            --yes \
+            --no-wait
+        return 0
+    fi
+
+    if [ "${platform}" = "tdx" ]; then
+        deploy_spec="-k "${tools_dir}/packaging/kata-deploy/kata-deploy/overlays/k3s""
+        cleanup_spec="-k "${tools_dir}/packaging/kata-deploy/kata-cleanup/overlays/k3s""
+    else
+        deploy_spec="-f "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml""
+        cleanup_spec="-f "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml""
+    fi
+
+    kubectl delete ${deploy_spec}
+    kubectl -n kube-system wait --timeout=10m --for=delete -l name=kata-deploy pod
+
+    sed -i -e "s|quay.io/kata-containers/kata-deploy:latest|${DOCKER_REGISTRY}/${DOCKER_IMAGE}:${DOCKER_TAG}|g" "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml"
+    cat "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml"
+    cat "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml" | grep "${DOCKER_REGISTRY}/${DOCKER_IMAGE}:${DOCKER_TAG}" || die "Failed to setup the tests image"
+    kubectl apply ${cleanup_spec}
+    sleep 180s
+
+    kubectl delete ${cleanup_spec}
+    kubectl delete -f "${tools_dir}/packaging/kata-deploy/kata-rbac/base/kata-rbac.yaml"
+    kubectl delete -f "${tools_dir}/packaging/kata-deploy/runtimeclasses/kata-runtimeClasses.yaml"
+}
+
+function main() {
+    action="${1:-}"
+
+    case "${action}" in
```

Comment:
what do you think of putting every case into a function?

Reply:
Done